addon/components/gavant-table.ts

```diff
@@ -0,0 +1,532 @@
+     * @memberof GavantTableComponent
+     */
+    debouncedDidResize() {
+        if (this.constrainColumnsToFit) {
+            //reset pan position on container resize
+            //TODO maybe eventually improve this to maintain pan position
+            //and dynamically show/hide as needed?
+            set(this, 'columnPanPosition', 0);
+            scheduleOnce('afterRender', this, 'updateColumnVisibility');
+        }
+    }
+
+    /**
+     * Hides and shows table columns depending on the available container width
+     *
+     * @memberof GavantTableComponent
+     */
+    updateColumnVisibility() {
+        const columns = this.columns || A();
+        const visibleColumns: NativeArray<TableColumn> = A();
+        const containerWidth = this.getElementWidth(this.element);
+        const allowFixedCols = containerWidth >= this.minFixedColTableWidth;
+        const panPosition = this.columnPanPosition;
+        let newTableWidth = 0;
+
+        for (const [i, col] of columns.entries()) {
+            let colIndex = allowFixedCols ? this.nonFixedColumns.indexOf(col) : i;
+            if ((col && col.isFixedLeft && allowFixedCols) || colIndex >= panPosition) {
+                let colWidth = col.staticWidth || 0;
+                let isVisible = (col.isFixedLeft && allowFixedCols) || newTableWidth + colWidth <= containerWidth;
+                if (isVisible) {
+                    newTableWidth += colWidth;
+                    visibleColumns.pushObject(col);
+                    // Prevent unwanted panning behavior on mobile that occurs with more than 2 columns
+                    const isMobile = get(get(this, 'media'), 'isMobile');
+                    if (isMobile && visibleColumns.length === 2) {
+                        break;
+                    }
+                } else {
+                    break;
+                }
+            }
+        }
+
+        setProperties(this, { containerWidth, visibleColumns });
+        this.changeDisplay();
+    }
+
+    /**
+     * Pans the table's visible column to the left or right the provided number of columns
+     * e.g. `-1` pans one column to the left, `2` pans two columns to the right
+     *
+     * @param {number} moveIndex -- A number indicating the number of columns to pan
+     * @memberof GavantTableComponent
+     */
+    panColumns(moveIndex: number) {
+        const newColumnPanPosition = this.columnPanPosition + moveIndex;
+        const pannedColumns = this.allowFixedCols ? this.nonFixedColumns : this.columns;
+        if (newColumnPanPosition < 0 || newColumnPanPosition >= pannedColumns.length) {
+            return;
+        }
+
+        set(this, 'columnPanPosition', newColumnPanPosition);
+        this.updateColumnVisibility();
+    }
+
+    /**
+     * Returns the computed width for the given element
+     *
+     * @param {Element} el -- A DOM element
+     * @returns {number}
+     * @memberof GavantTableComponent
+     */
+    getElementWidth(el: Element): number {
```

Comment:
This should probably be either A) made private, or b) made into a helper function that's imported by the table